### PR TITLE
feat(amazonq): add session configuration notification

### PR DESCRIPTION
**Pull request by zixlin7**
## Problem
need to add a notification to IdC users to contact their admin to configure extended session
## Solution
show the notification once to IdC users when their connection expired

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/util/authUtil.ts [hunk 1]
```diff
@@ -343,6 +346,30 @@ export class AuthUtil {
         }
     }
 
+    public async notifySessionConfiguration() {
+        const suppressId = 'amazonQSessionConfigurationMessage'
```
Comment (leigaol) [inline packages/core/src/codewhisperer/util/authUtil.ts:350]: This should be under typeof amazonQPromptName, see `type amazonQPromptName = keyof typeof amazonQPrompts`

Comment (leigaol) [reply packages/core/src/codewhisperer/util/authUtil.ts:350]: By doing so you don't have to do supressId as any

Comment (zixlin7) [reply packages/core/src/codewhisperer/util/authUtil.ts:350]: right I forgot to remove any type, done

**Review by leigaol — APPROVED** [review]

File: packages/core/src/codewhisperer/util/authUtil.ts [hunk 2]
```diff
@@ -345,6 +348,30 @@ export class AuthUtil {
...
+        const settings = AmazonQPromptSettings.instance
+        const shouldShow = await settings.isPromptEnabled(suppressId)
+        if (!shouldShow) {
+            return
+        }
+
+        const message = localize(
```
Comment (jpinkney-aws) [inline packages/core/src/codewhisperer/util/authUtil.ts:359]: This is probably a stupid question but will this show up to users who already have extended sessions on their IdC enabled?

Comment (zixlin7) [reply packages/core/src/codewhisperer/util/authUtil.ts:359]: yes it will, so we will only show it once as a FYI

Comment (jpinkney-aws) [reply packages/core/src/codewhisperer/util/authUtil.ts:359]: Can we detect if an IdC already has extended session enabled?

I'm guessing the team that owns the IdC might not be the team necessarily using this extension so having a dev (or multiple devs) message the IdC team to ask if extended sessions it enabled could annoy them

Comment (zixlin7) [reply packages/core/src/codewhisperer/util/authUtil.ts:359]: the session duration is controlled by the Q admin, product's idea is that users can contact their admin to configure the extended session if this is something that they want as end users

Comment (jpinkney-aws) [reply packages/core/src/codewhisperer/util/authUtil.ts:359]: Right, I was more concerned with notifying customers who already have extended session duration enabled. AFAICT we will also be notifying them and saying "hey tell your admin to enable extended session duration" even though its already enabled for them

Comment (zixlin7) [reply packages/core/src/codewhisperer/util/authUtil.ts:359]: agreed, but I don't think we need to get into checking end user's session configuration just for the notification purpose, it's something that they can easily check with their admin. This is what we aligned with product.

File: packages/core/src/codewhisperer/util/authUtil.ts [hunk 3]
```diff
@@ -345,6 +348,30 @@ export class AuthUtil {
...
+            'aws.amazonq.sessionConfiguration.message',
+            'Your maximum session length for Amazon Q can be extended to 90 days by your administrator. For more information, refer to How to extend the session duration for Amazon Q in the IDE in the IAM Identity Center User Guide.'
+        )
+
+        const learnMoreUrl = vscode.Uri.parse(
+            'https://docs.aws.amazon.com/singlesignon/latest/userguide/configure-user-session.html#90-day-extended-session-duration'
+        )
+        await vscode.window.showInformationMessage(message, localizedText.learnMore).then(async (resp) => {
```
Comment (hayemaxi) [inline packages/core/src/codewhisperer/util/authUtil.ts:367]:
```suggestion
        void vscode.window.showInformationMessage(message, localizedText.learnMore).then(async (resp) => {
```

Q activation will be blocked on the user interacting with this message (close, click a button). If they just ignore it or minimize the notifications then this await will never resolve.

File: packages/core/src/codewhisperer/util/authUtil.ts [hunk 4]
```diff
@@ -345,6 +348,30 @@ export class AuthUtil {
         }
     }
 
+    public async notifySessionConfiguration() {
```
Comment (hayemaxi) [inline packages/core/src/codewhisperer/util/authUtil.ts:351]: add telemetry?
[tookit_showNotification](https://github.com/aws/aws-toolkit-vscode/blob/2f5ff3a1e6304398437436819d6d16e21335eac6/packages/core/src/extensionNode.ts#L258)
[toolkit_invokeAction](https://github.com/aws/aws-toolkit-vscode/blob/2f5ff3a1e6304398437436819d6d16e21335eac6/packages/core/src/extensionNode.ts#L276)

**Review by jpinkney-aws — APPROVED** [review]

**Pull request merged**